e2e/ctl_v3_alarm_test.go

```diff
@@ -15,44 +15,62 @@
 package e2e
 
 import (
+	"os"
 	"strings"
 	"testing"
+	"time"
 )
 
-func TestCtlV3Alarm(t *testing.T) { testCtl(t, alarmTest, withQuota(64*1024)) }
+func TestCtlV3Alarm(t *testing.T) {
+	// The boltdb minimum working set is six pages.
+	testCtl(t, alarmTest, withQuota(int64(48*os.Getpagesize())))
+}
 
 func alarmTest(cx ctlCtx) {
 	// test small put still works
-	smallbuf := strings.Repeat("a", int(cx.quotaBackendBytes/100))
-	if err := ctlV3Put(cx, "abc", smallbuf, ""); err != nil {
+	smallbuf := strings.Repeat("a", 64)
+	if err := ctlV3Put(cx, "1st_test", smallbuf, ""); err != nil {
 		cx.t.Fatal(err)
 	}
 
-	// test big put (to be rejected, and trigger quota alarm)
-	bigbuf := strings.Repeat("a", int(cx.quotaBackendBytes))
-	if err := ctlV3Put(cx, "abc", bigbuf, ""); err != nil {
-		if !strings.Contains(err.Error(), "etcdserver: mvcc: database space exceeded") {
-			cx.t.Fatal(err)
+	// write some small chunks to fill up the database
+	buf := strings.Repeat("b", int(os.Getpagesize()/3))
+	for {
+		if err := ctlV3Put(cx, "2nd_test", buf, ""); err != nil {
+			if !strings.Contains(err.Error(), "etcdserver: mvcc: database space exceeded") {
+				cx.t.Fatal(err)
+			}
+			break
 		}
 	}
+
+	// quota alarm should now be on
 	if err := ctlV3Alarm(cx, "list", "alarm:NOSPACE"); err != nil {
 		cx.t.Fatal(err)
 	}
 
-	// alarm is on rejecting Puts and Txns
-	if err := ctlV3Put(cx, "def", smallbuf, ""); err != nil {
+	// check that Put is rejected when alarm is on
+	if err := ctlV3Put(cx, "3rd_test", smallbuf, ""); err != nil {
 		if !strings.Contains(err.Error(), "etcdserver: mvcc: database space exceeded") {
 			cx.t.Fatal(err)
 		}
 	}
 
+	// make some space
+	ctlV3Del(cx, []string{"2nd_test", "--prefix"}, 0)
```

Comment:
Since my change above loops and fills up the DB with puts, we need to free some space to do this last put.  Without freeing some space we can't put the small buffer.  There may be some combination of quota and buffer sizes that would work on some system, but that would not work in general.